instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/propagators/CompositeTextMapPropagator.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.propagators;
+
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.baggage.propagation.W3CBaggagePropagator;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import io.opentelemetry.extension.trace.propagation.B3Propagator;
+import io.opentelemetry.extension.trace.propagation.JaegerPropagator;
+import io.opentelemetry.extension.trace.propagation.OtTracePropagator;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.util.ClassUtils;
+
+/** Composite of {@link TextMapPropagator}. Defaults to W3C and BAGGAGE. */
+public final class CompositeTextMapPropagator implements TextMapPropagator {
+
+  private final Map<PropagationType, TextMapPropagator> mapping = new HashMap<>();
+  private final List<PropagationType> types;
+
+  public CompositeTextMapPropagator(BeanFactory beanFactory, List<PropagationType> types) {
+
+    this.types = types;
+
+    if (isOnClasspath("io.opentelemetry.extension.trace.propagation.B3Propagator")) {
+      mapping.put(
+          PropagationType.B3,
+          beanFactory
+              .getBeanProvider(B3Propagator.class)
+              .getIfAvailable(B3Propagator::injectingSingleHeader));
+    }
+    if (isOnClasspath("io.opentelemetry.extension.trace.propagation.JaegerPropagator")) {
+      mapping.put(
+          PropagationType.JAEGER,
+          beanFactory
+              .getBeanProvider(JaegerPropagator.class)
+              .getIfAvailable(JaegerPropagator::getInstance));
+    }
+    if (isOnClasspath("io.opentelemetry.extension.trace.propagation.OtTracerPropagator")) {
+      mapping.put(
+          PropagationType.OT_TRACER,
+          beanFactory
+              .getBeanProvider(OtTracePropagator.class)
+              .getIfAvailable(OtTracePropagator::getInstance));
+    }
+
+    mapping.put(PropagationType.W3C, W3CTraceContextPropagator.getInstance());
+    mapping.put(PropagationType.BAGGAGE, W3CBaggagePropagator.getInstance());
+    mapping.put(PropagationType.NOOP, TextMapPropagator.noop());
+  }
+
+  private static boolean isOnClasspath(String clazz) {
+    return ClassUtils.isPresent(clazz, null);
+  }
+
+  @Override
+  public Collection<String> fields() {
```

Comment:
Instead of implementing `fields()`, `inject()` and `extract()` yourself you can use `TextMapPropagator.composite()`; convert this class into some kind of factory class (or put the creation code in `PropagatorsConfiguration`), pass the configured propagators to the `composite()` method. WDYT?

Reply:
Sure, that's a way better approach. I can change `CompositeTextMapPropagator` to a factory class so as to not clutter the `PropagatorsConfiguration`.